clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -698,8 +701,21 @@ private KafkaConsumer(ConsumerConfig config,
                     config.getString(ConsumerConfig.ISOLATION_LEVEL_CONFIG).toUpperCase(Locale.ROOT));
             Sensor throttleTimeSensor = Fetcher.throttleTimeSensor(metrics, metricsRegistry.fetcherMetrics);
 
+            Sensor memoryPoolSensor = metrics.sensor("MemoryPoolUtilization");
+            MetricName memoryPoolDepletedPercentMetricName = metrics.metricName("memorypool-avg-depleted-percent",
+                    "consumer", "Percent of the time requests were not being read out of socket due to lack of memory");
+            memoryPoolSensor.add(memoryPoolDepletedPercentMetricName, new Rate(TimeUnit.MILLISECONDS));
+            long memoryPoolSize = config.getLong(ConsumerConfig.BUFFER_MEMORY_CONFIG);
+            int maxFetchBytes = config.getInt(ConsumerConfig.FETCH_MAX_BYTES_CONFIG);
+            if (memoryPoolSize <= maxFetchBytes) {
+                throw new ConfigException(ConsumerConfig.BUFFER_MEMORY_CONFIG + " should be greater than " + ConsumerConfig.FETCH_MAX_BYTES_CONFIG);
+            }
+            MemoryPool memoryPool = (memoryPoolSize > 0L)
+                    ? new SimpleMemoryPool(memoryPoolSize, maxFetchBytes, false, memoryPoolSensor)
+                    : MemoryPool.NONE;
```

Comment:
This wasn't documented on the kip. Maybe we could just throw an error if the size is less than 0?